### PR TITLE
Add "Save character" button

### PR DESCRIPTION
**Pull request by matthew-mcallister**
Added a save and delete button.

### CODE REVIEW

**Review by ltngonnguyen — APPROVED** [review]
Looking great, I was planning to make a similar function but seems like your implementation is a tad more thorough. Good work!

**Comment by oobabooga** [comment]
I like this but there are some issues:

1) Turn template only applies to instruct mode now
2) As of today, the name1/name2/context boxes for instruct and chat mode are different, so that all of them should be used as input and the appropriate boxes should be used in the function based on mode

File: modules/chat.py [hunk 1]
```diff
@@ -560,3 +561,27 @@ def upload_your_profile_picture(img, name1, name2, mode):
...
+        'name': name,
+        'greeting': greeting,
+        'context': context,
+        'turn_template': turn_template,
+    }
+    # Strip falsy
+    data = {k: v for k, v in data.items() if v is not None}
+
+    filepath = Path(f'{folder}/{name}.json')
+    filepath.write_text(json.dumps(data), encoding='utf-8')
```
[the remark this answers is not in the document]
Comment (oobabooga) [reply modules/chat.py:584]: yaml should definitely be used instead of json.

**Comment by matthew-mcallister** [comment]
Addressed comments and also added a delete button since the lack of one annoyed me.

Character loading currently doesn't support instruct mode any longer. Should that be implemented in this PR or can it be a separate PR?

**Comment by oobabooga** [comment]
I have opted to not include save/delete buttons for instruction templates because that seems to not be a routine task

**Pull request merged**